django/db/models/query.py

```diff
@@ -544,19 +545,31 @@ async def aiterator(self, chunk_size=2000):
         An asynchronous iterator over the results from applying this QuerySet
         to the database.
         """
-        if self._prefetch_related_lookups:
-            raise NotSupportedError(
-                "Using QuerySet.aiterator() after prefetch_related() is not supported."
-            )
         if chunk_size <= 0:
             raise ValueError("Chunk size must be strictly positive.")
         use_chunked_fetch = not connections[self.db].settings_dict.get(
             "DISABLE_SERVER_SIDE_CURSORS"
         )
+
+        chunk = deque()
+        num_items = 0
```

Comment:
What are the benefits of using a `deque` given we don't insert at `0` or bound the `deque` to `chunk_size`?

It seems we could use good ol' list here and possibly `len(chunk)` as well?

Reply:
When dealing with lists that have potentially thousands of entries (the default chunk size is 2000, and we commonly use a chunk size of 5000 for lengthy queries), I've been somewhat leery of the overhead involved in resizing lists.  I was also a little concerned about the cost of a len() function, although some quick googling a moment ago suggests that len() is an O(1) process.   Using len would certainly be cleaner, and using a list if we're not concerned about the overhead of reallocating list contents as the list grows.